unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/States/BuilderInWorldMode.cs

```diff
@@ -22,8 +23,10 @@ public class BuilderInWorldMode : MonoBehaviour
 
     public ActionController actionController;
 
-    public event System.Action OnInputDone;
-    public event System.Action<BuildInWorldCompleteAction> OnActionGenerated;
+    public event Action OnInputDone;
+    public event Action<BuildInWorldCompleteAction> OnActionGenerated;
+    public event Action<DCLBuilderInWorldEntity, bool> OnEntityDeselected;
+    public event Action OnEntitySelected;
```

Comment:
This action should be moved and called to the `BuilderInWorldEntityHandler`,  is his responsibility to handle this kind of selections

Reply:
Fixed